core/server/worker/src/main/java/alluxio/worker/block/AsyncCacheRequestManager.java

```diff
@@ -132,7 +138,14 @@ private boolean cacheBlockFromUfs(long blockId, long blockSize,
     try (BlockReader reader = mBlockWorker
         .readUfsBlock(Sessions.ASYNC_CACHE_SESSION_ID, blockId, 0)) {
       // Read the entire block, caching to block store will be handled internally in UFS block store
-      reader.read(0, blockSize);
+      // Note that, we read from UFS with a smaller buffer to avoid high pressure on heap
+      // memory when concurrent async requests are received and thus trigger GC.
+      long offset = 0;
+      while (offset < blockSize) {
+        long bufferSize = Math.min(8 * Constants.MB, blockSize - offset);
```

Comment:
Do you think it is worth making this buffer size configurable? Also I think we should consider using a pool of buffers to avoid creating many byte arrays.

Reply:
Agree. And it's better to extract this logic as a until method for future reuse.

Reply:
- I prefer not to expose another configurable buffer size---we already got too many and I don't see the importance of having this tunable by the users
- support to use a pool of buffers. this requires changing the impl of `alluxio.worker.block.UnderFileSystemBlockReader#read`, do you want to see it for 1.7.1?

Reply:
the current logic is quite special (read and discard) and I don't see an easy way to make an abstract that is useful without imposing overhead (e.g., allocating dst buffer)